phylanx/ast/parser/expression_def.hpp

```diff
@@ -140,7 +154,7 @@ namespace phylanx { namespace ast { namespace parser
             >>  raw[lexeme[(alpha | '_') >> *(alnum | '_')]]
             ;
 
-        string = lexeme['"' > raw[*(char_ - '"')] > '"']
+        string = lexeme['"' > *(unesc_char | "\\x" >> hex | (char_ - '"')) > '"']
```

Comment:
This will not limit the hex representation to maximally 2 digits, it will for instance accept `"\x1234"` and drop the higher order byte (`0x12`). I think this can be easily fixed by changing
```
qi::hex_type hex; 
```
to
```
qi::uint_parser<unsigned char, 16, 1, 2> hex;
```

Reply:
Done.